src/app/containers/SignupPage/index.js

```diff
@@ -43,25 +54,56 @@ function SignupPage({ history }) {
   const submit = React.useCallback(async () => {
     const sanitizedData = sanitize(data);
     setData(sanitizedData);
-    if (hasBlockingError(getErrors(sanitizedData))) return;
+    // eslint-disable-next-line no-shadow
+    if (hasBlockingError(getErrors(sanitizedData))) {
+      setPopupState(POPUP_STATE.ERROR);
+      return;
+    }
     setApiProgress(API_PROGRESS.REQ);
     // eslint-disable-next-line no-shadow
     const { data: response, error } = await signupWithData(sanitizedData);
 
     if (!!error || !response || !response.username) {
       // TODO: show popup
-      alert(`Đăng ký không thành công\n\n${error}`);
+      setPopupState(POPUP_STATE.ERROR);
       setApiProgress(API_PROGRESS.FAILED);
     } else {
       // TODO: show popup
-      alert(`Đăng ký thành công!`);
       setApiProgress(API_PROGRESS.SUCCESS);
-      !!history && history.push('/login');
+      setPopupState(POPUP_STATE.SUCCESS);
     }
   }, [data]);
 
+  const handleClosePopup = React.useCallback(() => {
+    setPopupState(POPUP_STATE.NONE);
+  });
+
+  const onButtonClick = React.useCallback(() => {
+    setRedirectState(true);
+  });
+
   return (
     <MainPanel.Container>
+      {redirectState === true ? <Redirect to="/login" /> : ''}
+      {popupState === POPUP_STATE.NONE ? (
+        ''
+      ) : popupState === POPUP_STATE.ERROR ? (
+        <Popup
+          onClose={handleClosePopup}
+          title="Đăng ký thất bại"
+          content="Đã xảy ra lỗi trong quá trình đăng ký. Vui lòng kiểm tra lại."
+          buttonText="OK"
+          variant="error"
+        />
+      ) : (
+        <Popup
+          title="Đăng ký thành công"
+          content="Cảm ơn bạn đã đăng ký tham gia FYT Code Cup! Vui lòng đăng nhập để tiếp tục."
+          buttonText="Đăng nhập"
+          variant="success"
+          onButtonClick={onButtonClick}
+        />
+      )}
```

Comment:
`POPUP_STATE.NONE` should be the last case because it's the default value. Readability is important.

```
{popupState === POPUP_STATE.ERROR ? (
...
) : popupState === POPUP_STATE.SUCCESS ? (
...
) : null}
```